samples: shell: shell_module: sysbuild.cmake fails with FATAL_ERROR on every non-Nordic board

## Description

Commit a3700684ff9d (\"samples: shell: shell_module: Extend with remote shell configuration\") added an unconditional check at the top of \`samples/subsys/shell/shell_module/sysbuild.cmake\`:

\`\`\`cmake
if(\"\${SB_CONFIG_REMOTE_BOARD}\" STREQUAL \"\")
  message(FATAL_ERROR \"REMOTE_BOARD must be set to a valid board name\")
endif()
\`\`\`

\`SB_CONFIG_REMOTE_BOARD\` is auto-defaulted in \`Kconfig.sysbuild\` for only a handful of Nordic targets (\`nrf5340/cpuapp\`, \`nrf5340bsim/cpuapp\`, \`nrf54h20/cpuapp\`, \`nrf54l15/cpuapp\`, \`nrf54lm20a/cpuapp\`). On every other board, the value is empty and configuration fails immediately as soon as sysbuild is involved.

The check is too strict: it fails even for variants that do not need the remote shell at all (\`sample.shell.shell_module\`, \`.minimal\`, \`.getopt\`, \`.login\`).

## Impact

Any board that mandates sysbuild for unrelated reasons (e.g. multi-image firmware assembly) cannot build the standalone shell sample. This currently blocks CI on PR #104035 (Corstone-1000-A320 board enablement) with 4 \`shell_module\` configurations erroring out.

## Reproducer

\`\`\`
west twister -p <any-non-Nordic-sysbuild-board> -T samples/subsys/shell/shell_module --build-only
\`\`\`

→ 4 errored: \`CMake Error at .../sysbuild.cmake:5 (message): REMOTE_BOARD must be set to a valid board name\`.

## Existing fix attempts

- **#108349** (\`samples: shell: shell_module: disable sysbuild by default for non-remote\`): sets \`common: sysbuild: false\` in \`sample.yaml\`. This addresses the case where sysbuild was being auto-enabled by twister, but **does not help** boards whose board yaml explicitly declares \`sysbuild: true\` — the unconditional \`FATAL_ERROR\` still fires.

- **#108455** (this issue's proposed fix): wraps \`ExternalZephyrProject_Add\` and its dependencies in a conditional so the remote build is skipped when \`REMOTE_BOARD\` is empty. Boards that opt in continue to get the remote app; others build the standalone shell sample as before.

The two are complementary: #108349 reduces unnecessary sysbuild invocations, #108455 makes the sample tolerant when sysbuild is mandated externally.

--- a/samples/subsys/shell/shell_module/sysbuild.cmake
+++ b/samples/subsys/shell/shell_module/sysbuild.cmake
@@ -1,19 +1,22 @@
 # Copyright (c) 2026 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-if("${SB_CONFIG_REMOTE_BOARD}" STREQUAL "")
-  message(FATAL_ERROR "REMOTE_BOARD must be set to a valid board name")
+# A remote build is only assembled when SB_CONFIG_REMOTE_BOARD has been set
+# (auto-defaulted for the Nordic targets in Kconfig.sysbuild, or set
+# explicitly by the user).  Boards that mandate sysbuild for unrelated
+# reasons (e.g. assembling a multi-image firmware bundle) but do not need
+# the remote shell get to build the standalone sample.
+if(NOT "${SB_CONFIG_REMOTE_BOARD}" STREQUAL "")
+  ExternalZephyrProject_Add(
+    APPLICATION remote
+    SOURCE_DIR ${APP_DIR}/remote
+    BOARD ${SB_CONFIG_REMOTE_BOARD}
+    BOARD_REVISION ${BOARD_REVISION}
+  )
+
+  sysbuild_add_dependencies(CONFIGURE ${DEFAULT_IMAGE} remote)
+  sysbuild_add_dependencies(FLASH ${DEFAULT_IMAGE} remote)
+
+  native_simulator_set_child_images(${DEFAULT_IMAGE} remote)
+  native_simulator_set_final_executable(${DEFAULT_IMAGE})
 endif()
-
-ExternalZephyrProject_Add(
-  APPLICATION remote
-  SOURCE_DIR ${APP_DIR}/remote
-  BOARD ${SB_CONFIG_REMOTE_BOARD}
-  BOARD_REVISION ${BOARD_REVISION}
-)
-
-sysbuild_add_dependencies(CONFIGURE ${DEFAULT_IMAGE} remote)
-sysbuild_add_dependencies(FLASH ${DEFAULT_IMAGE} remote)
-
-native_simulator_set_child_images(${DEFAULT_IMAGE} remote)
-native_simulator_set_final_executable(${DEFAULT_IMAGE})

